File: LeetCodeProblemsGo/LeetCode1304FindNUniqueIntegersSumUpToZero/LeetCode1304FindNUniqueIntegersSumUpToZero.go
package LeetCode1304FindNUniqueIntegersSumUpToZero

func sumZero(n int) []int {
	result := make([]int, 0, n)
	if n%2 == 1 {
		result = append(result, 0)
	}
	for i, e := 1, n/2; i <= e; i++ {
		result = append(result, i, -i)
	}
	return result
}

File: LeetCodeProblemsCpp/LeetCodeProblemsCpp/Problems/LeetCode1304FindNUniqueIntegersSumUpToZero.cpp
#include "LeetCode1304FindNUniqueIntegersSumUpToZero.h"

std::vector<int> LeetCode1304FindNUniqueIntegersSumUpToZero::sumZero(int n)
{
    std::vector<int> res{};
    res.reserve(n );
    if (n%2==1)
    {
        res.push_back(0);
    }
    for (int i=1;i<=n/2;i++)
    {
        res.push_back(i);
        res.push_back(-i);
    }
    return res;
}
